ioctl/doc/doc.go

```diff
@@ -38,6 +38,9 @@ func GenMarkdownTreeCustom(c *cobra.Command, dir string, name string, path strin
 		return err
 	}
 	defer func() {
+		if err != nil {
+			return
+		}
```

Comment:
if err != nil (for instance, write sth into file failed), we still need to call `f.Close()`

Reply:
fixed

Reply:
f.close()
return err

Reply:
f.close() should handle return error.
it's same with return f.close()

Reply:
```
defer func()
err1 := f.Close()
if err == nil {
    err = err1
}
```

Reply:
👍